sympy/codegen/tests/test_rewriting.py

```diff
@@ -218,7 +218,10 @@ def test_create_expand_pow_optimization():
     assert cc(x**4 - x**2) == '-x*x + x*x*x*x'
     i = Symbol('i', integer=True)
     assert cc(x**i - x**2) == 'pow(x, i) - x*x'
-
+    # gh issue 20753
+    cc2 = lambda x: ccode(optimize(x, [create_expand_pow_optimization(
+        4, base_req=lambda b: b.is_Function)]))
+    assert cc2(x**3 + sin(x)**3) == "pow(x, 3) + sin(x)*sin(x)*sin(x)"
```

Comment:
I'm not sure how codegen works but is it possible to do
```
y = sin(x)
z = y*y*y
```
Or is a `cse` style simplification applied later?

Reply:
I think that, currently, the users will have to apply `cse` themselves to get common subexpression elimination.

Reply:
Presumably they can't do that after this optimisation is applied though because the output has become a string of C code.

Reply:
Well, the conversion to C-code only happens in the test_*.py file. We could test against e
g. `srepr` if that's preferred.

Reply:
Oh, okay. I was just asking because I'm not familiar with the codegen module.